common/src/main/java/com/commercetools/sunrise/payment/domain/PaymentServiceProvider.java

```diff
@@ -0,0 +1,48 @@
+package com.commercetools.sunrise.payment.domain;
+
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.PaymentMethodInfo;
+import io.sphere.sdk.payments.PaymentStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A payment service provider provides functions that can be used to handle payment related tasks in
+ * a PSP specific way.
+ * @author mgatz
+ */
+public interface PaymentServiceProvider {
+
+    /**
+     * @return a unique identifier of the payment service provider
+     */
+    String getId();
+
+    /**
+     * @return a list of payment methods this PSP supports
+     */
+    List<PaymentMethodInfo> getAvailablePaymentMethods(); // TODO decide parameters
```

Comment:
I'm currently thinking about just passing a function that takes the list of all available methods and can then do any filtering the shop likes to (so the filter is out of scope for the payment framework)

As filtering is something 100% shop specific it should make sense to rely on the shop to provide a valid filter. The framework will provide a default "filter" that doesn't filter at all but that's much nicer than handling the parameter beeing null.